src/azure-cli-core/azure/cli/core/commands/parameters.py

```diff
@@ -493,3 +494,9 @@ def extra(self, dest, arg_type=None, **kwargs):
             merged_kwargs.pop('dest', None)
             self.command_loader.extra_argument_registry[self.command_scope][dest] = CLICommandArgument(
                 dest, **merged_kwargs)
+
+
+class TagUpdateOperation(str, Enum):
+    merge = "Merge"
+    replace = "Replace"
+    delete = "Delete"
```

Comment:
I don't think this should to be added in Azure CLI core. If the value appears in the request body, 

https://github.com/Azure/azure-cli/blob/7b109c5c98513022032d93337588a4b1ca6e230a/src/azure-cli/azure/cli/command_modules/resource/tests/latest/recordings/test_tag_update_resourcegroup.yaml#L56

it should be in the Swagger as a standalone `Enum` type. `operation` should support this `Enum` with `x-ms-enum`, similar to [`PropertyChangeType`](https://github.com/Azure/azure-rest-api-specs/blob/fd1212399155ecb17199b24434d780a42a6e5ec6/specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json#L5624) instead of only an inline `enum`:

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json#L5783-L5787

```py
          "enum": [
            "Replace",
            "Merge",
            "Delete"
          ]
```

Same applies to Python SDK.

https://github.com/Azure/azure-sdk-for-python/blob/8a322736b6588ff4b5e34796ac90eff56a8f3440/sdk/resources/azure-mgmt-resource/azure/mgmt/resource/resources/v2019_10_01/models/_models_py3.py#L1788-L1791

```py
    :param operation: The operation type for the patch API. Possible values
     include: 'Replace', 'Merge', 'Delete'
    :type operation: str or
     ~azure.mgmt.resource.resources.v2019_10_01.models.enum  # << This is wrong
```

Reply:
@jiasli, thanks for the detailed explanations. I have created a new [swagger pr](https://github.com/Azure/azure-rest-api-specs/pull/9828) to address it, and removed the validating logic for tag patch operation in cli. 

But this should be an improvement/refinement, and should not block our release (from my understanding, we don't need to wait for a new sdk release), right? We have some customers complaining about tags support for cli, and really need to catch the S171 release.